addons/wowl/static/tests/components/dropdown_tests.ts

```diff
@@ -0,0 +1,475 @@
+import * as owl from "@odoo/owl";
+import { OwlEvent } from "@odoo/owl/dist/types/core/owl_event";
+import QUnit from "qunit";
+import { Dropdown } from "../../src/components/dropdown/dropdown";
+import { DropdownItem } from "../../src/components/dropdown/dropdown_item";
+import { OdooEnv } from "../helpers";
+import { click, makeTestEnv, mount, nextTick } from "../helpers/utility";
+
+let env: OdooEnv;
+let parent: owl.Component;
+
+QUnit.module("Dropdown", {
+  async beforeEach() {
+    env = await makeTestEnv();
+  },
+  async afterEach() {
+    parent && parent.unmount();
+  },
+});
+
+QUnit.test("can be rendered", async (assert) => {
+  class Parent extends owl.Component {
+    static components = { Dropdown };
+    static template = owl.tags.xml`<Dropdown/>`;
+  }
+  parent = await mount(Parent, { env });
+  assert.containsOnce(parent.el!, "button.o_dropdown_toggler");
+  assert.containsNone(parent.el!, "ul.o_dropdown_menu");
+});
+
+QUnit.test("can be styled", async (assert) => {
+  class Parent extends owl.Component {
+    static components = { Dropdown, DropdownItem };
+    static template = owl.tags.xml`
+      <Dropdown class="one" togglerClass="'two'" menuClass="'three'">
+        <t t-set-slot="menu">
+          <DropdownItem class="four" />
+        </t>
+      </Dropdown>
+    `;
+  }
+  parent = await mount(Parent, { env });
+  await click(parent.el!, "button.o_dropdown_toggler");
+  assert.hasClass(parent.el!, "o_dropdown one");
+  const toggler = parent.el!.querySelector("button") as HTMLElement;
+  assert.hasClass(toggler, "o_dropdown_toggler two");
+  const menu = parent.el!.querySelector("ul") as HTMLElement;
+  assert.hasClass(menu, "o_dropdown_menu three");
+  const item = parent.el!.querySelector("li") as HTMLElement;
+  assert.hasClass(item, "four");
+});
+
+QUnit.test("menu can be toggled", async (assert) => {
+  class Parent extends owl.Component {
+    static components = { Dropdown };
+    static template = owl.tags.xml`<Dropdown/>`;
+  }
+  parent = await mount(Parent, { env });
+  await click(parent.el!, "button.o_dropdown_toggler");
+  assert.containsOnce(parent.el!, "ul.o_dropdown_menu");
+  await click(parent.el!, "button.o_dropdown_toggler");
+  assert.containsNone(parent.el!, "ul.o_dropdown_menu");
+});
+
+QUnit.test("initial open state can be true", async (assert) => {
+  class Parent extends owl.Component {
+    static components = { Dropdown };
+    static template = owl.tags.xml`<Dropdown open="true"/>`;
```

Comment:
I thought we decided to rename this props

Reply:
That was not clear, I thought we decided this props was well documented :-D
What do you suggest: "startOpen" ?

Reply:
ok then, keep it as it is

Reply:
Hmmm finally I just changed it to "startOpen"...
I think this is better than just "open".

"open" would let people think that they may control the opening state from outside.
"startOpen" tells exactly what it does